### PR TITLE
[IA-5060] Inherit manager roles to notebook-cluster and persistent-disk

### PR DESCRIPTION
**Pull request by rtitle**
Ticket: https://broadworkbench.atlassian.net/browse/IA-5060

What:

Phase 1 of the [Simplify Leo Access Control](https://broadworkbench.atlassian.net/browse/IA-5059) epic.

This PR: 
* Allows `notebook-cluster` and `perstistent-disk` resource types to inherit from `google-project` or workspace.
* Makes `notebook-cluster/connect` and `persistent-disk/attach` actions auth domain constrainable.

Why:

We are trying to clean up/simplify Leonardo access control logic. Part of this is standardizing on consistent Sam resource types and using inheritance to propagate permissions. This PR sets up the Sam model; subsequent Leonardo changes will be made to migrate to it.

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've filled out the [Security Risk Assessment](https://sdarq.dsp-appsec.broadinstitute.org/jira-ticket-risk-assesment) (requires Broad Internal network access) and attached the result to the JIRA ticket

### CODE REVIEW

File: src/main/resources/reference.conf [hunk 1]
```diff
@@ -803,22 +814,30 @@ resourceTypes = {
...
+      }
     }
     reuseIds = false
+    authDomainConstrainable = true
   }
   persistent-disk = {
     actionPatterns = {
       read = {
-        description = "read metadata and contents of persistent disk"
+        description = "read metadata of persistent disk"
```
Comment (rtitle) [inline src/main/resources/reference.conf:835]: `read` is just used for listing disks; `attach` is needed to actually view the contents of a disk

File: src/main/resources/reference.conf [hunk 2]
```diff
@@ -790,6 +800,7 @@ resourceTypes = {
       }
       connect = {
         description = "connect to the Jupyter notebook running on the notebook cluster"
+        authDomainConstrainable = true
```
Comment (LizBaldo) [inline src/main/resources/reference.conf:803]: I am a bit confused about how granular the authDomainConstrainable is here. I thought that every resources / actions would inherit that configuration from the workspace after Doug's PR

Comment (rtitle) [reply src/main/resources/reference.conf:803]: My understanding was you still have to annotate which actions on the child are authDomainConstrainable (like `connect`). @dvoet can you confirm?

File: src/main/resources/reference.conf [hunk 3]
```diff
@@ -803,22 +814,30 @@ resourceTypes = {
...
     roles = {
       creator = {
         roleActions = ["status", "connect", "delete", "read_policies", "stop_start", "modify"]
       }
+      manager = {
+        roleActions = ["status", "delete", "read_policies"]
+      }
     }
     reuseIds = false
+    authDomainConstrainable = true
```
Comment (rtitle) [inline src/main/resources/reference.conf]: I blindly followed this pattern from other resources -- does `authDomainConstrainable = true` need to be specified here at the resource level, in addition to the action level?

**Comment by rtitle** [comment]
VERY draft Leo PR to start creating Sam resources with parents (phase 2): https://github.com/DataBiosphere/leonardo/pull/4781/

I'll try to test these together next week.

File: src/main/resources/reference.conf (hunk 2 above)
Comment (dvoet) [reply src/main/resources/reference.conf:803]: @rtitle is correct. The auth domain is inherited but which actions have the auth domain applied is controlled by the `authDomainConstrainable` setting.

File: src/main/resources/reference.conf (hunk 3 above)
Comment (dvoet) [reply src/main/resources/reference.conf]: hmmm... interesting. Looking at the code, [only the action level setting is used](https://github.com/broadinstitute/sam/blob/1cf4c65577fc7d321fb68bda447e08e059e3f5c5/src/main/scala/org/broadinstitute/dsde/workbench/sam/model/SamModel.scala#L142).

**Review by dvoet — APPROVED** [review]

File: src/main/resources/reference.conf (hunk 3 above)
Comment (rtitle) [reply src/main/resources/reference.conf]: Interesting… I'll just set it at the action level then I think.

**Comment by rtitle** [comment]
@LizBaldo requested review on this PR. I tested on a BEE and verified that all the resources (runtime, disk, app) continue to work normally.

**Review by LizBaldo — APPROVED** [review]

**Review by tlangs — APPROVED** [review]

**Comment by ungwudik** [comment]
@rtitle or anyone here - Is the failure in `orch-swat-test-job (dev, qa)` a known issue or flakiness?

**Comment by rtitle** [comment]
@ungwudik I noticed that, the Orch test failed with:
```
[info] - should link an eRA Commons account with access to none of the supported closed-access datasets *** FAILED *** (4 seconds, 225 milliseconds)
[info]   org.broadinstitute.dsde.workbench.service.RestException: {"causes":[],"message":"<!DOCTYPE html>\n<html lang=\"en\">\n<head>\n<meta charset=\"utf-8\">\n<title>Error</title>\n</head>\n<body>\n<pre>Cannot GET /dev/public-key.pem</pre>\n</body>\n</html>\n","source":"shibboleth","stackTrace":[],"statusCode":404}
```
Looks like the Orch swat test failed to get a dev shibboleth endpoint. This PR didn't touch anything related to orch/shibboleth. I'm not sure if this is a known dev infra issue? (cc @tlangs)

**Pull request merged**